dace/transformation/dataflow/vectorization.py

```diff
@@ -1,13 +1,138 @@
 # Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
 """ Contains classes that implement the vectorization transformation. """
-from dace import data, dtypes, registry, symbolic, subsets
-from dace.sdfg import nodes, SDFG, SDFGState, propagation
+from re import M
+import re
+from numpy import core
+import sympy
+
+from numpy.lib.arraysetops import isin
+from dace import data, dtypes, registry, symbolic, subsets, symbol
+from dace.frontend.octave.lexer import raise_exception
+from dace.sdfg import nodes, SDFG, SDFGState, propagation, InterstateEdge
 from dace.sdfg import utils as sdutil
 from dace.sdfg.scope import ScopeSubgraphView
 from dace.transformation import transformation
 from dace.transformation.helpers import replicate_scope
-from dace.properties import Property, make_properties
+from dace.properties import Property, make_properties, EnumProperty, SymbolicProperty
 import itertools
+import dace.transformation.dataflow.vectorization_infer_types as infer_types
+import dace.sdfg.analysis.vector_inference as vector_inference
+import dace.codegen.targets.sve.util as sve_util
+import dace.frontend.operations
+
+
+# Find the innermost state in which the node is
+# There is probably a better solution than this
+def get_innermost_state_for_node(sdfg: SDFG, node):
+    for s in sdfg.states():
+        for n, state in s.all_nodes_recursive():
+            if n == node:
+                return state
+
+    raise Exception("State for the node {n} not found".format(n=node))
+
+
+# Find the innermost sdfg in which the node is
+# There is probably a better solution than this
+def get_innermost_sdfg_for_node(sdfg: SDFG, node):
+    for s in sdfg.all_sdfgs_recursive():  #sdfg
+        for n in s.nodes():  # states
+            for i in n.nodes():  # nodes
+                if node == i:
+                    return s
+    raise Exception("sdfg for the node {n} not found".format(n=node))
+
+
+# Find the innermost sdfg in which the array is
+# There is probably a better solution than this
+def get_innermost_sdfg_for_array(sdfg: SDFG, array):
+    for s in sdfg.all_sdfgs_recursive():  #sdfg
+        if array in s.arrays:
+            return s
+    raise Exception("sdfg for the array {n} not found".format(n=array))
+
+
+def collect_maps_to_vectorize(sdfg: SDFG, state, map_entry):
+    """
+    Collect all maps and the corresponding data descriptors that have to be vectorized
+    if target == FPGA.
+    """
+    # Collect all possible and maps
+    all_maps_entries_exits = set()
+
+    for n, s in sdfg.all_nodes_recursive():
```

Comment:
Maybe add an explanation why we look through all states and nodes, even though we get a state and map entry as input (I assume because we _have_ to vectorize everything on the same time because we change the data type).